amida-tech/indaba-client
# Feature/INBA-494 Subject delete checks in wizard

#### What's this PR do?
Add subject delete checks and prompts when deleting subjects from the project wizard

#### Related JIRA tickets:
[DS-494](https://jira.amida-tech.com/browse/INBA-494)

#### How should this be manually tested?
1. Ensure you have a project with a subject. Let us call this existing subject "A"
1. Create a new project through the wizard
1. Add subject "A", and a new subject that no other project has
1. Delete subject "A", and check that you are prompted first, and that the `isDeleted` flag is set in the corresponding ProductUOA row, but NOT the UnitOfAnalysis for "A".
1. Delete the new subject that no other project has, and check that you are prompted first, and that the `isDeleted` column is set in both the ProductUOA table AND the UnitOfAnalysis table

#### Any background context you want to provide?
This branch sits on #245, review that first for a smaller changeset

#### Screenshots (if appropriate):
